lambda-notes/src/components/Form.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+class Form extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      id: null,
+      tags: [],
+      title: '',
+      textBody: '',
+    }
+  }
+
+  componentDidMount(){
+    if (this.props.note) {
+      this.setState({
+        id: this.props.note._id,
+        tags: this.props.note.tags,
+        title: this.props.note.title,
+        textBody: this.props.note.textBody,
+      })
+    }
+  }
+
+  handleSubmit = event => {
+    event.preventDefault();
+    const note = {
+      ...this.state
+    }
+    if (this.props.note){
+      this.props.submit(note);
+      this.props.history.push(`/${this.props.match.params.id}`);
+    } else {
+      this.props.submit(note);
+      this.props.history.push('/');
```

Comment:
this is causing a race condition where submitting the new note is causing to dispatch an action that will switch the boolean savingNotes to true which is causing the ListView component to have conflicts instantiating fetchingNotes() in render & CDM

React is giving you the following warning : 
Warning: Cannot update during an existing state transition (such as within `render`). Render methods should be a pure function of props and state.